### PR TITLE
[Conf]Make default_storage_medium configurable

### PR DESCRIPTION
**Pull request by WingsGo**
For #2954

### CODE REVIEW

**Comment by morningman** [comment]
I don't quite understand how this configuration solves the problem you mentioned in the #2954 ? For example, the current clusters are HDD, and now some machines need to be replaced with SSDs. How to use this configuration in this case?

**Comment by WingsGo** [comment]
> I don't quite understand how this configuration solves the problem you mentioned in the #2954 ? For example, the current clusters are HDD, and now some machines need to be replaced with SSDs. How to use this configuration in this case?

This configuration can not change partitions already is HDD，if cluster medium is changed from HDD to SSD, admin should change the history partitions storage medium. But for users, they may have no idea that the cluster storage_medium is changed or even they don't konw about they can specify the storage_medium.so in this case, they will create table as usual, this will cause all partitions's storage_medium after migration is default (HDD), but currently, the cluster have no machine who's medium is HDD, this will cause some problem about cluster balance.

**Comment by morningman** [comment]
So you mean, when all the disks of a cluster are replaced with SSD, we set this parameter to SSD?

Then if only a part of the disks in the cluster are replaced with SSDs, do we need to change this parameter to SSDs? If it is changed, then all new data will fall on the SSD?

**Comment by morningman** [comment]
So you mean, when all the disks of a cluster are replaced with SSD, we set this parameter to SSD?

Then if only a part of the disks in the cluster are replaced with SSDs, do we need to change this parameter to SSDs? If it is changed, then all new data will fall on the SSD?

**Comment by WingsGo** [comment]
In my opinion, this configuration gives admin a choice, if part of the disk are replaced with SSD, we don't need to change anything, the PR doesn't affect anything either. This PR is just an admin option, it will help admin control the system more conviently without change users's behavior. If the cluster is combine with HDD and SSD, admin should let users konwn what kind of table's medium should they choose.

File: fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ public void write(DataOutput out) throws IOException {
         out.writeInt(idToDataProperty.size());
         for (Map.Entry<Long, DataProperty> entry : idToDataProperty.entrySet()) {
             out.writeLong(entry.getKey());
-            if (entry.getValue() == DataProperty.DEFAULT_HDD_DATA_PROPERTY) {
+            if (entry.getValue() == DataProperty.DEFAULT_DATA_PROPERTY) {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java]: Here will be a problem.
For now, `DataProperty.DEFAULT_DATA_PROPERTY` is configuration, so here, it is all possible to write `true` or `false`. And when reading, `true` may point to HDD, may point to SSD, depends on the config.

**Review by morningman — CHANGES_REQUESTED** [review]

File: fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java (hunk 1 above)
Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java]: I am confused why we should add a judge whether the DataProperty is default, can we just write `entry.getValue().write(out);`  and read as `DataProperty.read(in)`. I think the idToDataProperty's values can't be null?

Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java]: If we need to compatible with old version, I can chnage it to `entry.getValue() == new DataProperty(TStorageMedium.HDD)`

Comment (morningman) [reply fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java]: You are righ, it is a little bit confused..
You can change it as you said.

File: fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java [hunk 2]
```diff
@@ -170,7 +170,7 @@ public String toString() {
 
         for (Map.Entry<Long, DataProperty> entry : idToDataProperty.entrySet()) {
             buff.append(entry.getKey()).append("is HDD: ");;
-            if (entry.getValue() == DataProperty.DEFAULT_HDD_DATA_PROPERTY) {
+            if (entry.getValue() == DataProperty.DEFAULT_DATA_PROPERTY) {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java]: Change the log print `is HDD: `

Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java]: done

File: fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java (hunk 1 above)
Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/catalog/PartitionInfo.java]: done

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**